oss-internship-2020/libuv/CMakeLists.txt

```diff
@@ -0,0 +1,394 @@
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.16)
+
+project(libuv_sandbox)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+file(MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/uv_wrapper")
+add_custom_command(
+
+  OUTPUT "${CMAKE_BINARY_DIR}/uv_wrapper/uv_wrapper.h"
+         "${CMAKE_BINARY_DIR}/uv_wrapper/uv_wrapper.cc"
+
+  WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+
+  COMMAND python3 "generator/wrapper_generator.py"
+                  "libuv/include/uv.h"
+                  "${CMAKE_BINARY_DIR}/uv_wrapper/uv_wrapper.h"
+                  "${CMAKE_BINARY_DIR}/uv_wrapper/uv_wrapper.cc"
+
+  COMMENT "Generate the wrapper header and source files"
+
+)
+
+option(UV_SAPI_ENABLE_EXAMPLES "" ON)
```

Comment:
Can you swith the `UV_` prefix around? I.e., use `SAPI_UV_ENABLE_EXAMPLES`, etc.